gtk-vnc 1.2.0

cho-m opened the pull request:
Created with `brew bump-formula-pr`.

carlocab commented:
Patch appears to have been merged? Can't find the commit that applies it though.

cho-m commented:
Yup, fix is in v1.2.0: https://gitlab.gnome.org/GNOME/gtk-vnc/-/commit/94377d8716d3b4c7e78d763ac074fa937415fb44

cho-m commented:
Fixed CI failure, which was due to `"-Dwith-vala=false"` needing to be changed to `"-Dwith-vala=disabled"`

Also added 2 patches from local build attempt:
1. Issue when disabling vala, which is fixed in: https://gitlab.gnome.org/GNOME/gtk-vnc/-/merge_requests/14
2. Issue due to `implicit declaration of function 'GDK_IS_QUARTZ_DISPLAY'`.
   This still fails if I use `-Wno-implicit-function-declaration` due to missing symbol.
   It appears to be a side-effect of removing fall-back behavior for older Gtk2 (https://gitlab.gnome.org/GNOME/gtk-vnc/-/commit/78225889ce861c0d3ea4aa97603fef252817505f#f8968bfcf81a5d2be650c5ae8a62786a3992db1f).
   Based fix off of https://gitlab.freedesktop.org/spice/spice-gtk/blob/master/src/vncdisplaykeymap.c#L95

cho-m commented on Formula/gtk-vnc.rb:
```diff
@@ -1,8 +1,9 @@
-  sha256 "a81a1f1a79ad4618027628ffac27d3391524c063d9411c7a36a5ec3380e6c080"
+  url "https://download.gnome.org/sources/gtk-vnc/1.2/gtk-vnc-1.2.0.tar.xz"
+  sha256 "7aaf80040d47134a963742fb6c94e970fcb6bf52dc975d7ae542b2ef5f34b94a"
+  license "LGPL-2.1-or-later"
```
Based on license info I saw:
- `meson.build` has `license: 'LGPLv2.1+'`
- `COPYING.LIB` (modified 9 years ago) is for LGPL-2.1
- `Changelog` on 2017-08-08 mentions switching LGPL from v2-only to v2-or-later
- Code in `src/` shows LGPLv2.0 or later:
    ```
     * modify it under the terms of the GNU Lesser General Public
     * License as published by the Free Software Foundation; either
     * version 2.0 of the License, or (at your option) any later version.
    ```

carlocab commented on Formula/gtk-vnc.rb:
```diff
@@ -22,12 +23,20 @@ class GtkVnc < Formula
+
+  # Fix compile failure in src/vncdisplaykeymap.c
+  # error: implicit declaration of function 'GDK_IS_QUARTZ_DISPLAY' is invalid in C99
   patch :DATA
```
Has this been reported?

cho-m replied:
Not yet. I'll see if I have time today to create GNOME account and open an issue.

carlocab replied:
You can sign in with your GitHub account.

cho-m replied:
Opened https://gitlab.gnome.org/GNOME/gtk-vnc/-/issues/16

carlocab replied:
Let's add that to the comment in a syntax-only PR when this PR is merged.

carlocab replied:
#78019
